geom/src/polygon.rs

```diff
@@ -476,19 +476,37 @@ fn to_geo(pts: &Vec<Pt2D>) -> geo::Polygon<f64> {
     )
 }
 
-fn from_geo(p: geo::Polygon<f64>) -> Polygon {
-    Polygon::buggy_new(
-        p.into_inner()
-            .0
-            .into_points()
-            .into_iter()
-            .map(|pt| Pt2D::new(pt.x(), pt.y()))
-            .collect(),
-    )
+impl From<geo::Polygon<f64>> for Polygon {
+    fn from(poly: geo::Polygon<f64>) -> Self {
+        let (exterior, interiors) = poly.into_inner();
+        Polygon::with_holes(
+            Ring::from(exterior),
+            interiors.into_iter().map(Ring::from).collect(),
+        )
+    }
+}
+
+impl From<Polygon> for geo::Polygon<f64> {
+    fn from(poly: Polygon) -> Self {
+        if let Some(mut rings) = poly.rings {
+            let exterior = rings.pop().expect("expected poly.rings[0] to be exterior");
+            let interiors: Vec<geo::LineString<f64>> =
+                rings.into_iter().map(geo::LineString::from).collect();
+            Self::new(exterior.into(), interiors)
+        } else {
+            let exterior_coords = poly
```

Comment:
Note this will break for a polygon produced by `polyline.make_polygons`, for instance, because `points` won't form a ring. It looks like on the [geo side](https://docs.rs/geo/0.16.0/geo/struct.Polygon.html#method.new), it'll `pts.push(pts[0])` on input if needed. That'll still turn out to be nonsense for this, since the points will go back and forth.

I think this is fine for now. Ideally every `geom::Polygon` will properly have an exterior ring. aka, we have to whittle down the callers of `Polygon::buggy_new` and `Polygon::precomputed`. Some of those will be hard, maybe impossible (like triangulated polygons coming from `lyon`).